proto/cmp/services/activity/v1alpha/activity_types.proto

```diff
@@ -0,0 +1,235 @@
+
+}
+
+message ActivityProductInfoDetails {
+  // Context for Inventory system concepts that need to be included in an info response,
+  // like an OwnerCode, PTC_OfferParameters, Tax codes, Disclosure RefID, etc. or a
+  // serialized combination of these codes.
+  string context = 1;
+
+  // Errors encountered in the generation of the response. Standardized error list
+  // to be created.
+  repeated string errors = 2;
+
+  // Warnings encountered in the generation of the response. This section might include
+  // information about fields not processed due to incompatibilities with the
+  // inventory system
+  repeated string warnings = 3;
+
+  // Supplier code from the Inventory System for this activity response.
+  string supplier_unit_code = 4;
+
+  // An identifier for external sessions, aiding in tracking and continuity across
+  // sessions.
+  string external_session_id = 5;
+
+  // Activity Category Code
+  string category_code = 6;
+
+  // Category name which describes the category_code
+  string category_name = 7;
+
+  // Type Code
+  string type_code = 8;
+
+  // Type name which describes type_code
+  string type_name = 9;
+
+  // Supplier specific code, matching the supplier code in ProductList and ProductInfo
+  // messages
+  cmp.types.v1alpha.SupplierProductCode supplier_code = 10;
+
+  // Supplier Name
+  string supplier_code_name = 11;
+
+  // product_code
+  cmp.types.v1alpha.ProductCode product_code = 12;
+
+  // Duration range (min, max) of the activity in minutes
+  cmp.types.v1alpha.DurationRange duration_range = 13;
+
+  // Time given for confirmation before the offer expires
+  cmp.types.v1alpha.Duration max_confirmation_duration = 14; 
+
+  // Allow Free Sale
+  bool allow_free_sale = 15;
+
+  // The ability to confirm instantly
+  bool instant_confirmation = 16;
+
+  // Can be delivered instantly
+  bool instant_delivery = 17;
+
+  // Check availability
+  bool availability_required = 18;
+  
+  // Availability Type
+  string availability_type = 19;
+
+  // Delivery Formats; such as QR Code, NFT, Ticket, ... etc
+  repeated string delivery_formats = 20;
+
+  // Delivery Methods; such as Email, SMS, Post, ... etc
+  repeated string delivery_methods = 21;
+
+  // Redemption method
+  string redemption_method = 22;
+
+}
+
+
+message ActivityUnitInfo {
+  // Schedule
+  cmp.types.v1alpha.DateTimeRange schedule = 1; 
+
+  // Description of the activity unit
+  string descriptive_text = 2;
+
+  // Unit Code
+  string unit_code = 3;
+
+  // Unit Code Description
+  string unit_code_description = 4;
+
+}
+
+
+message ActivityServiceInfo {
+  // Description of the Service Info
+  string descriptive_text = 1;
+  
+  // Service Code
+  string service_code = 2;
+
+  // Service Code Description
+  string service_code_description = 3;
+
+  // Included items in the activity service
+  repeated string included = 4;
+
+  // Encluded items in the activity service
+  repeated string excluded = 5;
```

Comment:
```suggestion
message ActivityServiceInfo {
  // Description of the Service Info
  string description = 1;
  
  // Service Code
  string code = 2;

  // Service Code Description
  string code_description = 3;

  // Included items in the activity service
  repeated string included = 4;

  // Encluded items in the activity service
  repeated string excluded = 5;
```

Same here, instead of repeating the message's name in the fields, I think we should just use the field name like `code`, `description` etc.

Also check other message through out the PR.

Reply:
Same solution: lets change the first "description" (was "descriptive_text") to: "additional_info" and move it to 3rd position